controllers/oauth_client_test.go

```diff
@@ -0,0 +1,69 @@
+package controllers_test
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	. "go-google-scraper-challenge/helpers/test"
+	"go-google-scraper-challenge/initializers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("OAuthClientController", func() {
+	Describe("GET /oauth_client", func() {
+		It("renders with status 200", func() {
+			response := MakeRequest("GET", "/oauth_client", nil)
+
+			Expect(response.StatusCode).To(Equal(http.StatusOK))
+		})
+	})
+
+	Describe("POST /oauth_client", func() {
+		It("redirects to oauth client detail page", func() {
+			form := url.Values{}
+			body := strings.NewReader(form.Encode())
+			response := MakeRequest("POST", "/oauth_client", body)
+			currentPath := GetCurrentPath(response)
+
+			Expect(response.StatusCode).To(Equal(http.StatusFound))
+			Expect(currentPath.Path).To(MatchRegexp(`\/oauth_client\/\b`))
+		})
+
+		It("sets the success message", func() {
+			form := url.Values{}
+			body := strings.NewReader(form.Encode())
+			response := MakeRequest("POST", "/oauth_client", body)
+			flash := GetFlashMessage(response.Cookies())
+
+			Expect(flash.Data["success"]).To(Equal("The Client was successfully created"))
+			Expect(flash.Data["error"]).To(BeEmpty())
+		})
+	})
+
+	Describe("GET /oauth_client/:id", func() {
+		Context("given valid client id", func() {
```

Comment:
Should we have the `given invalid client id` case?

Reply:
@gutakk ✅  added on [53f8080](https://github.com/carryall/go-google-scraper-challenge/pull/28/commits/53f8080cf9df86ab2ddc1d345d42eaa94b9f7908)